src/jit/morph.cpp

```diff
@@ -6953,6 +6958,10 @@ GenTreePtr          Compiler::fgMorphCall(GenTreeCall* call)
         {
             szFailReason = "Localloc used";
         }
+        else if ((call->gtFlags & GTF_CALL_NO_RETURN) != 0)
+        {
+            szFailReason = "No return";
+        }
```

Comment:
The tail call code is part of the epilog but with my change no epilog is generated anymore because the BB is converted to `BBJ_THROW`. The net result is that a test failed because an exception was no longer being thrown.